### PR TITLE
Add tier-3 support for powerpc64 and riscv64 openbsd

### PR DESCRIPTION
**Pull request by semarie**
# powerpc64
- MCP for [powerpc64-unknown-openbsd tier-3 support](https://github.com/rust-lang/compiler-team/issues/551)
- only need to add spec definition in rustc_target

# riscv64
- MCP for [riscv64-unknown-openbsd tier-3 support](https://github.com/rust-lang/compiler-team/issues/552)
- add spec definition in rustc_target
- follow freebsd about avoiding linking with `libatomic`

### CODE REVIEW

**Comment by petrochenkov** [comment]
Tier 3 target policy (https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html#tier-3-target-policy) says that MCP is not necessary for such targets, so I think I can just r+ this.

File: compiler/rustc_target/src/spec/powerpc64_unknown_openbsd.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use crate::abi::Endian;
+use crate::spec::{LinkerFlavor, Target, TargetOptions};
+
+pub fn target() -> Target {
+    let mut base = super::openbsd_base::opts();
+    base.cpu = "ppc64".into();
+    base.pre_link_args.entry(LinkerFlavor::Gcc).or_default().push("-m64".into());
```
Comment (petrochenkov) [inline compiler/rustc_target/src/spec/powerpc64_unknown_openbsd.rs:7]:
```suggestion
    base.add_pre_link_args(LinkerFlavor::Gcc, &["-m64"]);
```

**Comment by petrochenkov** [comment]
r=me with the comment addressed.
@rustbot author

**Comment by semarie** [comment]
just a note to point [zulip conversation](https://rust-lang.zulipchat.com/#narrow/stream/233931-t-compiler.2Fmajor-changes/topic/riscv64-unknown-openbsd.20tier-3.20support.20compiler-team.23552/near/295364232) where it stands that the target name `riscv64-unknown-openbsd` should match linux counterparts (even if it is OpenBSD) and be called `riscv64gc-unknown-openbsd`.

even if I disagree, I will look to address that.

**Pull request merged**